CONTRIBUTING.md

```diff
@@ -33,109 +41,210 @@ For an introduction to the basics of reST syntax, the source format in which Spy
+Pre-commit itself is installed with the above command, and the hooks should be enabled by running the following from the root of this repo:
+
+```shell
+pre-commit install --hook-type pre-commit --hook-type commit-msg
 ```
 
 The hooks will be automatically run against any new/changed files every time you commit.
 It may take a few minutes to install the needed packages the first time you commit, but subsequent runs should only take a few seconds.
-If you made one or more commits before installing the hooks (not recommended), to run them manually on all the files in the repo, execute:
+If you made one or more commits before installing the hooks (not recommended), you can run them manually on all the files in the repo with:
 
-```bash
+```shell
 pre-commit run --all-files
 ```
 
-**Note**: Most of the hooks fix the problems they detect automatically (the hook output will say ``files were modified by this hook``, but no errors/warnings will be listed), but they will still abort the commit so you can double-check everything first.
-Once you're satisfied, ``git add .`` again and commit again.
+**Note**: Many of the hooks fix the problems they detect automatically (the hook output will say ``Files were modified by this hook``, but no errors/warnings will be listed), but they will still abort the commit so you can double-check everything first.
+Once you're satisfied, ``git add .`` and commit again.
 
 
-### Build the docs
 
-To build the docs locally with Sphinx, you can easily do so with our makefile from the Terminal/command line (or the Anaconda prompt on Windows).
+## Building the docs
 
-To build just the docs for the current version, run the following if on macOS/Linux:
+To build the docs locally with Sphinx, you can easily do so with our makefile from the Terminal/command line, or via running the appropriate Sphinx command.
+
+To build just the docs for the current version, run the following on a system with ``make`` (macOS/Linux):
 
 ```bash
 make docs
 make serve
 ```
 
-or, if on Windows:
+To build the full site with the documentation for all Spyder versions, run:
 
-```cmd
-cd doc
-make.bat html
-cd ..
+```bash
+make multidocs
 ```
 
-**NOTE:** You can try to use the `make` commands on Windows if you have `make` installed and something like `Cygwin`, `MSYS2` or `MYSYS`, or by using Windows Subsystem for Linux (`WSL`).
+On a system without ``make`` (like Windows, by default), or to build the docs manually, run:
 
-To build the full site with the documentation for all Spyder versions, run:
+```shell
+python -m sphinx -n -W --keep-going -b html doc doc/_build/html
+```
 
-```bash
-make multidocs
+If not opened automatically with ``make serve``, you should be able to view the HTML output inside the resulting ``_build/html`` directory the build commands produce; ``index.html`` is the main page.
```

Comment:
```suggestion
If the documentation is not opened automatically after running ``make serve``, you can view the output with your web browser.
Specifically, you need to go to the ``_build/html`` directory and open ``index.html``, which is the main page of our docs.
```

This sentence is very confusing to me, although I understand what you mean by it. I tried to simplify it, but I don't know if you'll agree with my suggestion.

Reply:
I see what you mean and agree with your suggestion overall, and clarified further the language particularly around ``make serve`` (since the issue isn't that it will be run but won't work, but that it won't be run at all since users don't have ``make`` or just don't use it) and added some other clarifications.

```suggestion
If you've run ``make serve``, the rendered documentation should open automatically in your default web browser.
Otherwise, navigate to the ``_build/html`` directory inside the ``spyder-docs`` repository and open ``index.html`` (the main page of the docs) in your browser.
```